module/VuFind/src/VuFind/Db/Service/UserResourceService.php

```diff
@@ -107,35 +142,27 @@ public function createOrUpdateLink(
         UserListEntityInterface|int $listOrId,
         string $notes = ''
     ): UserResourceEntityInterface {
-        $resource = $resourceOrId instanceof ResourceEntityInterface
-            ? $resourceOrId : $this->getDbService(ResourceServiceInterface::class)->getResourceById($resourceOrId);
-        if (!$resource) {
-            throw new Exception("Cannot retrieve resource $resourceOrId");
-        }
-        $list = $listOrId instanceof UserListEntityInterface
-            ? $listOrId : $this->getDbService(UserListServiceInterface::class)->getUserListById($listOrId);
-        if (!$list) {
-            throw new Exception("Cannot retrieve list $listOrId");
-        }
-        $user = $userOrId instanceof UserEntityInterface
-            ? $userOrId : $this->getDbService(UserServiceInterface::class)->getUserById($userOrId);
-        if (!$user) {
-            throw new Exception("Cannot retrieve user $userOrId");
-        }
-        $params = [
-            'resource_id' => $resource->getId(),
-            'list_id' => $list->getId(),
-            'user_id' => $user->getId(),
-        ];
-        if (!($result = $this->getDbTable('UserResource')->select($params)->current())) {
+        $resource = $this->getDoctrineReference(ResourceEntityInterface::class, $resourceOrId);
+        $user = $this->getDoctrineReference(UserEntityInterface::class, $userOrId);
+        $list = $this->getDoctrineReference(UserListEntityInterface::class, $listOrId);
+        $params = compact('resource', 'list', 'user');
+        $result = current($this->entityManager->getRepository(UserResourceEntityInterface::class)
+            ->findBy($params));
+
+        if (empty($result)) {
             $result = $this->createEntity()
                 ->setResource($resource)
                 ->setUser($user)
                 ->setUserList($list);
         }
         // Update the notes:
         $result->setNotes($notes);
-        $this->persistEntity($result);
+        try {
+            $this->persistEntity($result);
+        } catch (\Exception $e) {
+            $this->logError('Could not save user resource: ' . $e->getMessage());
+            return false;
```

Comment:
Also here is a false value returned, but in function, the return type is expecting : UserResourceEntityInterface

Reply:
Good catch -- we really don't need that exception handling at all. I have removed it in 9e9aba1a74fbeee77a07f64549341fe0d32b6534.